datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -352,7 +352,11 @@ public void run() {
     } finally {
       _stoppedLatch.countDown();
       if (null != _consumer) {
-        _consumer.close();
+        try {
+          _consumer.close();
```

Comment:
should we have retries for this?

Reply:
I don’t think retry is necessary since we are anyways on the shutdown path. Let me know if you have a reason for why you think retry may be helpful

Reply:
synced with Sonam offline, it looks like assign can be called on same set of partitions. And since the task will be shut down, we won't use the consumer for polling again even if its not closed.